sycl/test/host-interop-task/host-task-dependency.cpp

```diff
@@ -0,0 +1,178 @@
+  S::buffer<int, 1> Buf2;
+  S::buffer<int, 1> Buf3;
+  std::mutex Mutex;
+  std::condition_variable CV;
+};
+
+void Thread1Fn(Context &Ctx) {
+  // 0. initialize resulting buffer with apriori wrong result
+  {
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::host_buffer>
+        Acc(Ctx.Buf2);
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx)
+      Acc[Idx] = -1;
+  }
+
+  // 1. submit task writing to buffer 1
+  Ctx.Queue.submit([&](S::handler &CGH) {
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::global_buffer>
+        GeneratorAcc(Ctx.Buf1, CGH);
+
+    auto GeneratorKernel = [GeneratorAcc]() {
+      for (size_t Idx = 0; Idx < GeneratorAcc.get_count(); ++Idx)
+        GeneratorAcc[Idx] = Idx;
+    };
+
+    CGH.single_task<class GeneratorTask>(GeneratorKernel);
+  });
+
+  // 2. submit host task writing from buf 1 to buf 2
+  auto HostTaskEvent = Ctx.Queue.submit([&](S::handler &CGH) {
+    S::accessor<int, 1, S::access::mode::read,
+                S::access::target::host_buffer>
+        CopierSrcAcc(Ctx.Buf1, CGH);
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::host_buffer>
+        CopierDstAcc(Ctx.Buf2, CGH);
+
+    auto CopierHostTask = [CopierSrcAcc, CopierDstAcc, &Ctx]() {
+      for (size_t Idx = 0; Idx < CopierDstAcc.get_count(); ++Idx)
+        CopierDstAcc[Idx] = CopierSrcAcc[Idx];
+
+      bool Expected = false;
+      bool Desired = true;
+      assert(Ctx.Flag.compare_exchange_strong(Expected, Desired));
+
+      // let's employ some locking here
+      {
+        std::lock_guard<std::mutex> Lock(Ctx.Mutex);
+        Ctx.CV.notify_all();
+      }
+    };
+
+    CGH.codeplay_host_task(CopierHostTask);
+  });
+
+  // 3. submit simple task to move data between two buffers
+  Ctx.Queue.submit([&](S::handler &CGH) {
+    S::accessor<int, 1, S::access::mode::read,
+                S::access::target::global_buffer>
+        SrcAcc(Ctx.Buf2, CGH);
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::global_buffer>
+        DstAcc(Ctx.Buf3, CGH);
+
+    CGH.depends_on(HostTaskEvent);
+
+    auto CopierKernel = [SrcAcc, DstAcc]() {
+      for (size_t Idx = 0; Idx < DstAcc.get_count(); ++Idx)
+        DstAcc[Idx] = SrcAcc[Idx];
+    };
+
+    CGH.single_task<class CopierTask>(CopierKernel);
+  });
+
+  // 4. check data in buffer #3
+  {
+    S::accessor<int, 1, S::access::mode::read,
+                S::access::target::host_buffer>
+        Acc(Ctx.Buf3);
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx)
+      assert(Acc[Idx] == Idx && "Invalid data in third buffer");
+  }
+}
+
+void Thread2Fn(Context &Ctx) {
+  std::unique_lock<std::mutex> Lock(Ctx.Mutex);
+
+  // T2.1. Wait until flag F is set eq true.
+  Ctx.CV.wait(Lock, [&Ctx] { return Ctx.Flag.load(); });
+
+  assert(Ctx.Flag.load());
+
+  // T2.2. print some "hello, world" message
+  Ctx.Message = "Hello, world";
```

Comment:
What are you trying to test with this message? From what I can see, you may as well use a bool?